text/0000-compiler-team-trusted-contrib-maintainer.md

```diff
@@ -0,0 +1,416 @@
+their contributions to gain employment/contracts to contribute to the project
+full-time (or otherwise), the naming of the compiler team contributor role can
+be confusing. An employer unfamiliar with the project may not realise that a
+compiler team contributor is a role within the project which recognises regular
+contribution and trust rather than just having made a handful of contributions
+and thus being a contributor.
+
+[rfc2689]: https://rust-lang.github.io/rfcs/2689-compiler-team-contributors.html
+[review_queue_analysis]: https://borrowed.dev/p/on-the-compiler-teams-review-queue
+
+# Definitions
+[definitions]: #definitions
+
+There are various permissions/privileges/responsibilities which will be
+referenced in later sections of this RFC, defined here:
+
+- **r+**
+  - Contributors with *r+* privilege are able to approve pull requests to be
+    merged by *bors*. Contributors should not merge their own pull requests (with
+    the exception of re-approving their own work on behalf of another contributor
+    after a rebase or similarly trivial change). *r+* permissions apply to the
+    whole repository, but [it is expected][expectations] that contributors limit
+    themselves to only those parts of the *rust-lang/rust* repository that are under
+    the purview of the compiler team (unless granted *r+* from other teams too), and
+    for subsystems/pull requests that they are confident reviewing.
+- **try**
+  - Contributors with *try* permissions are able to trigger complete toolchain
+    builds for a pull request or commit, which are then used by *rustc-perf* and
+    *crater*. *try* permissions aren't available to everyone because try builds
+    can pose a security risk: try builds have access to secrets and the resulting
+    builds are hosted on `static.rust-lang.org` where we would never want
+    malicious code.
+- **review rotation**
+  - Contributors on the review rotation will be randomly assigned to new pull
+    requests submitted to the compiler. Being on the review rotation is one of the
+    best ways for contributors to help the compiler team and learn new parts of
+    the compiler. Review capacity is one of the most important resources that the
+    team has, as it enables our progress in the compiler's continued development
+    and maintenance.
+- **organization membership**
+  - Contributors that are added to the *rust-lang/compiler* team in the GitHub
+    organization can be assigned to issues/pull requests, modify labels, receive
+    group mentions and receive a "Member" badge next to their name.
+- **rustc-perf**
+  - Contributors with permissions to use *rustc-perf* can request benchmarking
+    of their pull requests (and pull requests they are reviewing). *rustc-perf*
+    permissions are useful for regular contributors as it is common to need to
+    request benchmarks from contributors with permissions. *rustc-perf* permissions
+    only make sense alongside *try* permissions.
```

Comment:
What is the reason to have this ability be separate from `try`? Is it purely to give further flexibility in case it is necessary; or are there additional security implications to `@perfbot` that need to be taken into the account that `@bors try` doesn't need to worry about?

Reply:
Just flexibility